dwave/cloud/api/client.py

```diff
@@ -298,10 +302,79 @@ class DWaveAPIClient:
     def __init__(self, **config):
         self.config = {}
         for opt, default in self.DEFAULTS.items():
-            self.config[opt] = config.get(opt, default)
+            val = config.get(opt)
+            if val is None:
+                val = default
+            self.config[opt] = val
+
+        logger.debug(f"{type(self).__name__} initialized with config={self.config!r}")
 
         self.session = self._create_session(self.config)
 
+    @classmethod
+    def from_client_config(cls, config: ClientConfig, **kwargs):
+        """Create class instance based on
+        :class:`~dwave.cloud.config.models.ClientConfig` config.
+        """
+        logger.trace(f"{cls.__name__}.from_client_config(config={config!r}, **{kwargs!r})")
+
+        if config.headers:
+            headers = config.headers.copy()
+        else:
+            headers = {}
+        if config.connection_close:
+            headers.update({'Connection': 'close'})
+
+        opts = dict(
+            endpoint=config.endpoint,
+            token=config.token,
+            cert=config.cert,
+            headers=headers,
+            timeout=config.request_timeout,
+            retry=config.request_retry.model_dump(),
+            proxies=dict(http=config.proxy, https=config.proxy),
+            verify=not config.permissive_ssl,
+        )
+
+        # context-sensitive config update
+        update_config(opts, kwargs)
+
+        return cls(**opts)
+
+    @classmethod
+    def from_config_file(cls, config_file: Optional[str] = None,
+                         profile: Optional[str] = None, **kwargs):
+        """Create class instance based on config file / environment / kwarg options
+        (same format as :class:`~dwave.cloud.client.base.Client`).
+        """
+        logger.trace(f"{cls.__name__}.from_config_file("
+                     f"config_file={config_file!r}, profile={profile!r}, **{kwargs!r})")
+
+        options = load_config(config_file=config_file, profile=profile, **kwargs)
+        config = validate_config_v1(options)
+        return cls.from_client_config(config)
+
+    @classmethod
+    def from_config(cls, config: Optional[Union[ClientConfig, str]] = None, **kwargs):
+        """Create class instance based on client config or config file.
+
+        Args:
+            config:
+                Path to configuration file or client config model. Based on `config`
+                type, we dispatch object creation to either :meth:`.from_config_file`
+                or :meth:`.from_client_config`. If omitted, config load from file is
+                attempted.
```

Comment:
"config load from file" --> can we just add "(the default)" to "to either :meth:`.from_config_file`" in the previous sentence (otherwise "If omitted, attempts to load configuration from file")

Okay, now I'll try not to look at more docstrings